### PR TITLE
ECL-97: Added tax enrolments connector

### PR DESCRIPTION
**Pull request by TonyCOgilvie**

### CODE REVIEW

File: app/uk/gov/hmrc/economiccrimelevyregistration/connectors/TaxEnrolmentsConnector.scala [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class TaxEnrolmentsConnector @Inject() (appConfig: AppConfig, httpClient: HttpClient)(implicit ec: ExecutionContext) {
+
+  private val taxEnrolmentsUrl: String =
+    s"${appConfig.taxEnrolmentsBaseUrl}/tax-enrolments/service/$serviceName/enrolment"
+
+  def enrol(taxEnrolment: CreateEnrolmentRequest)(implicit hc: HeaderCarrier): Future[Unit] =
```
Comment (peter-hazell) [inline app/uk/gov/hmrc/economiccrimelevyregistration/connectors/TaxEnrolmentsConnector.scala:34]: minor thing but lets rename the value to `createEnrolmentRequest` for consistency

File: test/uk/gov/hmrc/economiccrimelevyregistration/connectors/TaxEnrolmentsConnectorSpec.scala [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import uk.gov.hmrc.economiccrimelevyregistration.models.eacd.CreateEnrolmentRequest.serviceName
+
+import scala.concurrent.Future
+
+class TaxEnrolmentsConnectorSpec extends SpecBase {
+  val mockHttpClient: HttpClient = mock[HttpClient]
+  val connector                  = new TaxEnrolmentsConnector(appConfig, mockHttpClient)
+  val taxEnrolmentsUrl           = s"${appConfig.taxEnrolmentsBaseUrl}/tax-enrolments/service/$serviceName/enrolment"
+
+  "upsertEnrolment" should {
```
Comment (peter-hazell) [inline test/uk/gov/hmrc/economiccrimelevyregistration/connectors/TaxEnrolmentsConnectorSpec.scala:34]:
```suggestion
  "enrol" should {
```

File: test/uk/gov/hmrc/economiccrimelevyregistration/connectors/TaxEnrolmentsConnectorSpec.scala [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import uk.gov.hmrc.http.{HttpClient, HttpResponse}
+import com.danielasfregola.randomdatagenerator.RandomDataGenerator.derivedArbitrary
+import uk.gov.hmrc.economiccrimelevyregistration.models.eacd.CreateEnrolmentRequest.serviceName
+
+import scala.concurrent.Future
+
+class TaxEnrolmentsConnectorSpec extends SpecBase {
+  val mockHttpClient: HttpClient = mock[HttpClient]
+  val connector                  = new TaxEnrolmentsConnector(appConfig, mockHttpClient)
+  val taxEnrolmentsUrl           = s"${appConfig.taxEnrolmentsBaseUrl}/tax-enrolments/service/$serviceName/enrolment"
```
Comment (peter-hazell) [inline test/uk/gov/hmrc/economiccrimelevyregistration/connectors/TaxEnrolmentsConnectorSpec.scala:32]: can probably just inline this in the test as there is only one test?

File: app/uk/gov/hmrc/economiccrimelevyregistration/connectors/TaxEnrolmentsConnector.scala (hunk 1 above)
Comment (TonyCOgilvie) [reply app/uk/gov/hmrc/economiccrimelevyregistration/connectors/TaxEnrolmentsConnector.scala:34]: Done

File: test/uk/gov/hmrc/economiccrimelevyregistration/connectors/TaxEnrolmentsConnectorSpec.scala (hunk 3 above)
Comment (TonyCOgilvie) [reply test/uk/gov/hmrc/economiccrimelevyregistration/connectors/TaxEnrolmentsConnectorSpec.scala:32]: Done

**Pull request merged**